microsoft-edge/progressive-web-apps-chromium/how-to/digital-goods-api.md

```diff
@@ -32,53 +32,113 @@ See:
 ## Checking whether the Digital Goods API is available
 
-To detect whether you've correctly enabled the API on your website, check for the `getDigitalGoodsService` method in the window object:
+To detect whether you've correctly enabled the API on your website by using your origin trial token, check whether the `getDigitalGoodsService` method exists on the `window` object:
 
 ```javascript
 if ('getDigitalGoodsService' in window) {
-  // Digital Goods API is supported!
+  // The Digital Goods API is supported.
 } else {
   console.log('DigitalGoodsService is not available.');
-  // Use other payment method
+  // Use another payment method.
 }
 ```
 
+<!-- todo: 
+See also:
+* [getDigitalGoodsService in window object]()
+-->
+
 
 <!-- ====================================================================== -->
-## Connecting to the Microsoft Store Billing service (`getDigitalGoodsService` method)
+## Connecting to the Microsoft Store Billing service (`window.getDigitalGoodsService` method)
+
+<!-- todo: what type is returned by window.getDigitalGoodsService("https://store.microsoft.com/billing") ? -->
 
-Use the `getDigitalGoodsService` method to connect to the Microsoft Store Billing service.
+Use the `getDigitalGoodsService` method of the `window` object to connect to the Microsoft Store Billing service.
 
 The Digital Goods API was designed to be compatible with various browsers and digital stores, similar to how the Payment Request API is browser-agnostic and can be used with different payment providers.  To retrieve an instance of the service for Microsoft Store Billing, pass the string `"https://store.microsoft.com/billing"` as the payment method to the `getDigitalGoodsService` method.
 
 If the method throws an error, the Microsoft Store Billing payment method is not available (such as when the user is accessing your PWA through the browser).  Alternatively, consider providing a different payment method for transactions.
 
 ```javascript
 if (window.getDigitalGoodsService === undefined) {
-  // Digital Goods API is not supported in this context.
+  // The Digital Goods API isn't supported in this context.
   return;
 }
 try {
   const digitalGoodsService = await window.getDigitalGoodsService("https://store.microsoft.com/billing");
   // Use the service here.
   ...
 } catch (error) {
-  // Our preferred service provider is not available.
+  // The preferred service provider is not available.
   // Use a web-based payment flow instead.
   return;
 }
 ```
 
+This payment method `getDigitalGoodsService("https://store.microsoft.com/billing")` is available only for a PWA that's installed from the Microsoft Store, on Windows.  No other settings are needed.
+
+<!-- todo: 
+See also:
+* [window.getDigitalGoodsService]()
+-->
+
 
 <!-- ====================================================================== -->
 ## Querying item details (`getDetails` method)
 
+<!-- todo: what type is the getDetails method on? -->
```

Comment:
what type is each method on?  especially, what type is returned by:
window.getDigitalGoodsService("https://store.microsoft.com/billing")

@runyuany

Reply:
We can add a ref: [digitalgoodsservice](https://wicg.github.io/digital-goods/#digitalgoodsservice)
The WICG doc listed all method return types.

Reply:
@mikehoffms because this still an experimental API, and because it will eventually be documented on MDN, we don't need to write the entire reference documentation for the API, and it's ok to link to the spec that @runyuany metnioned. I'll leave it to you to decide what needs to be documented here absolutely, vs. what we can just link to the spec.

Reply:
When saying to use a certain method, it's necessary to state what type or object or interface or class that method is on, so I'm adding that, eg 
```
Use the `consume` method of the `DigitalGoodsService` interface to consume a purchase.
...
* [consume() method](https://wicg.github.io/digital-goods/#consume-method) of the `DigitalGoodsService` 
interface, in _Digital Goods API: Draft Community Group Report_.
```